src/compute/src/logging/compute.rs

```diff
@@ -347,16 +347,23 @@ pub fn construct<A: Allocate>(
 
         let frontier_delay = frontier_delay
             .as_collection()
+            .explode_one(|(dataflow, source_id, worker, delay)| {
+                (
+                    (dataflow, source_id, worker, delay.next_power_of_two()),
+                    (i128::try_from(delay).expect("delay too big"), 1),
+                )
+            })
```

Comment:
As mentioned offline, I think you can remove this `explode_one` by emitting the correct data directly.